web/src/components/profile/ProfileAddress.test.tsx

```diff
@@ -0,0 +1,824 @@
+import { fireEvent, render, screen, within } from "@testing-library/react";
+
+const Config = getMergedConfig();
+
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("@mui/material", () => mockMaterialUI());
+
+function mockMaterialUI(): typeof materialUi {
+  return {
+    ...jest.requireActual("@mui/material"),
+    useMediaQuery: jest.fn(),
+  };
+}
+
+describe("<ProfileAddress  />", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  const renderNjBusinessComponent = ({ municipalities }: { municipalities?: Municipality[] }): void => {
+    render(
+      <ThemeProvider theme={createTheme()}>
+        <WithStatefulAddressData
+          initialData={emptyAddressData}
+          municipalities={municipalities || [generateMunicipality({})]}
+        >
+          <ProfileAddress businessLocationType="NJ" />
+        </WithStatefulAddressData>
+      </ThemeProvider>
+    );
+  };
+
+  const renderNonNjUSBusinessComponent = (): void => {
+    render(
+      <ThemeProvider theme={createTheme()}>
+        <WithStatefulAddressData initialData={emptyAddressData} municipalities={[generateMunicipality({})]}>
+          <ProfileAddress businessLocationType="US" />
+        </WithStatefulAddressData>
+      </ThemeProvider>
+    );
+  };
+
+  const renderIntlBusinessComponent = (): void => {
+    render(
+      <ThemeProvider theme={createTheme()}>
+        <WithStatefulAddressData initialData={emptyAddressData} municipalities={[generateMunicipality({})]}>
+          <ProfileAddress businessLocationType="INTL" />
+        </WithStatefulAddressData>
+      </ThemeProvider>
+    );
+  };
+
+  describe("NJ business location", () => {
+    it("renders profile address line 1 field with input value", () => {
+      const address = generateAddress({
+        addressLine1: "1111 Home Alone",
+      });
+      useMockBusiness(
+        generateBusiness({
+          formationData: generateFormationData({
+            completedFilingPayment: false,
+            formationFormData: generateFormationFormData({ ...address }),
+          }),
+        })
+      );
+      renderNjBusinessComponent({});
+      expect(screen.getByLabelText("Address line1")).toHaveValue("1111 Home Alone");
+    });
+
+    it("renders profile address line 1 reflects changed value", () => {
+      const address = generateAddress({
+        addressLine1: "",
+      });
+      useMockBusiness(
+        generateBusiness({
+          formationData: generateFormationData({
+            completedFilingPayment: false,
+            formationFormData: generateFormationFormData({ ...address }),
+          }),
+        })
+      );
+      renderNjBusinessComponent({});
+
+      expect(screen.getByLabelText("Address line1")).toHaveValue("");
+      fillText("Address line1", "1111 Home Alone");
+      expect(screen.getByLabelText("Address line1")).toHaveValue("1111 Home Alone");
+    });
+
+    it("renders profile address line 1 error", () => {
```

Comment:
```suggestion
    it("displays an error on address line 1 when field is blurred while empty", () => {
```

This seems to align a bit closer with what is in the test.